### ISSUE TITLE
Use the new Framework API

### ISSUE THREAD

**Issue by jonniebigodes** [storybookjs/addon-google-analytics#4]
Opening this issue so we can address the upcoming changes on how addons are built/updated for the upcoming 7.0 release. If anyone is interested in making the changes required and continuing to support and maintain the addon, let us know so that we're aware and help/guide the community moving forward. To help with this process, we've built an abridged guide [here](https://chromatic-ui.notion.site/Addon-migration-for-Storybook-7-002f2f627c1d470298ca94894ab8a040), and should any questions or issues, please reach out to us in the prerelease channel in our [Discord Server](https://discord.gg/storybook).

**Comment by fgaleano** [comment]
I'm just stumbling into issues with this addon and I think I'd like to volunteer to address the issues I've found so far. That I know of:

1. The latest version does not work with Google Universal Analytics.
2. ReactGA also does not support UA. But `react-ga4` exist which indeed supports GUA.
3. The latest version does not work on Storybook 7 Beta, mainly due to changes in the `addons` APIs.

What I *think* needs to be done is the following:

1. For item 1 and 2, simply upgrade to `react-ga4` and update a couple of API calls that changed with `react-ga4`. Based on what I see on the source code, I think this is the easiest part, only a few lines of code plus any required tests.
2. For item 3, upgrading the project dependencies to `next` _might_ be all that's needed. the `api.on()` API seems to still be there and work the same way. But I'm not 100% sure if the current code will be compatible. I don't see much reliance on the addon API other than `register` and `.on()`.

@jonniebigodes Does that sound like close enough summary? If so, I could get started on later this week.

Thanks.

**Comment by jonniebigodes** [comment]
@fgaleano, thanks for following up with me on this issue and your summary of all your findings. Greatly appreciate it. If you're still interested in working on them and updating the addon, we'd welcome it 🙇 . 

Looking forward to seeing the pull request.

Hope you have a great week.

Stay safe

**Comment by fgaleano** [comment]
@jonniebigodes Yes, I can follow up and submit this work. I did a first pass this weekend and was able to get the tracking tag working with GA4 and the latest Storybook version. I now need to test on `next`. I don't expect any big issues.

Happy to get this assigned to me to prevent any duplicate work. I'll submit the PR in the next couple of days for initial review.

**Comment by jonniebigodes** [comment]
@fgaleano that's great to hear. I already assigned you to it and we're looking forward to seeing the pull request.

Hope you have a fantastic day. 

Stay safe

**Comment by fgaleano** [comment]
@jonniebigodes I submitted 2 PRs. One for v6.x to add support for GA4. This is needed for anyone still not upgrading to v7 but wanting to use the latest GA version.

The second PR is for SB7 support. Unfortunately, I'm not able to get it to work because I'm running into an error during build time. I added some details in that PR. Hopefully someone from the team can point me in the right direction.

Thanks.

**Comment by jonniebigodes** [comment]
@fgaleano I saw them and already answered in the 7.0 one. I will follow up with the team and see how we can unblock you to get this one merged in light of the issues you've encountered migrating to the upcoming release. As already mentioned in the said pull request, expect a bit of a delay as the team is gearing up for the release, and there are some items that require attention. Do you think that will work for you?

**Comment by fgaleano** [comment]
@jonniebigodes Yeah, no worries. I understand the team is busy. I implemented a temporary "creative" solution on my project while the official release gets underway. Thank you!
